src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -21,93 +24,85 @@ public QueryTimeouts(IDocumentStore documentStore, string endpointName)
             TriggerCleanupEvery = TimeSpan.FromMinutes(2);
             CleanupGapFromTimeslice = TimeSpan.FromMinutes(1);
             shutdownTokenSource = new CancellationTokenSource();
+            logger = LogManager.GetLogger<QueryTimeouts>();
         }
 
         public TimeSpan CleanupGapFromTimeslice { get; set; }
         public TimeSpan TriggerCleanupEvery { get; set; }
+        public Func<DateTime> GetUtcNow { get; set; } = () => DateTime.UtcNow;
 
         public async Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
         {
-            var now = DateTime.UtcNow;
+            var now = GetUtcNow();
```

Comment:
I don't "get" private getters. You should be able to look at something's state after you mutate it. Or check its state before you decide if you want to mutate it.

As far as the GetUtcNow, my intention is to be able to add tests at a later time which would tightly control `now` so that we can build tests that take stale indexes into account. Maybe one day, figure out a way to query timeouts that doesn't require the Cleanup concept.

So in both cases (but for different reasons) I'd prefer to tell ReSharper to take a hike there.